src/meepgeom.cpp

```diff
@@ -350,12 +350,20 @@ meep::realnum matgrid_val(vector3 p, geom_box_tree tp, int oi, material_data *md
     ++matgrid_val_count;
   }
 
-  return (md->material_grid_kinds == material_data::U_MIN
-              ? umin
-              : (md->material_grid_kinds == material_data::U_PROD
-                     ? uprod
-                     : (md->material_grid_kinds == material_data::U_SUM ? usum / matgrid_val_count
-                                                                        : udefault)));
+  meep::realnum u_interp = (md->material_grid_kinds == material_data::U_MIN
+                            ? umin
+                            : (md->material_grid_kinds == material_data::U_PROD
+                               ? uprod
+                               : (md->material_grid_kinds == material_data::U_SUM ? usum / matgrid_val_count
+                                  : udefault)));
+
+  // project interpolated grid point
+  meep::realnum u_proj;
+  if (md->beta)
```

Comment:
I would explicitly check `md->beta != 0` for clarity.

Reply:
Fixed.